iceoryx_posh/source/roudi/application/roudi_app.cpp

```diff
@@ -67,37 +65,28 @@ void RouDiApp::registerSigHandler() noexcept
     sigemptyset(&act.sa_mask);
     act.sa_handler = roudiSigHandler;
     act.sa_flags = 0;
-    if (-1 == sigaction(SIGINT, &act, NULL))
+    if (cxx::makeSmartC(sigaction, cxx::ReturnMode::PRE_DEFINED_SUCCESS_CODE, {0}, {}, SIGINT, &act, nullptr)
+            .hasErrors())
     {
-        LogError() << "Calling sigaction() failed";
-        std::terminate();
+        LogFatal() << "Calling sigaction() failed";
+        errorHandler(Error::kROUDI_APP__COULD_NOT_REGISTER_SIGNALS, nullptr, ErrorLevel::FATAL);
```

Comment:
Please add a return after the `errorHandler`. Otherwise it is possible that the execution continues when the errorHandler does not terminate when the user overrides the default behavior.

Reply:
I don't get that. How will a `return` help here if `errorHandler()` does not return?